scripts/extensions-emulator-tests/tests.ts

```diff
@@ -1,62 +1,103 @@
+  cleanUpExtensionsCache();
+  fs.mkdirSync(process.env.FIREBASE_EXTENSIONS_CACHE_PATH);
+}
+
+function cleanUpExtensionsCache(): void {
+  if (
+    process.env.FIREBASE_EXTENSIONS_CACHE_PATH &&
+    fs.existsSync(process.env.FIREBASE_EXTENSIONS_CACHE_PATH)
+  ) {
+    rimraf.sync(process.env.FIREBASE_EXTENSIONS_CACHE_PATH);
+  }
+}
 
 function readConfig(): FrameworkOptions {
-  const filename = path.join(EXTENSION_ROOT, "test-firebase.json");
+  const filename = path.join(__dirname, "firebase.json");
   const data = fs.readFileSync(filename, "utf8");
   return JSON.parse(data);
 }
 
-describe("extension emulator", () => {
+function logIncludes(msg: string) {
+  return (data: unknown) => {
+    if (typeof data !== "string" && !Buffer.isBuffer(data)) {
+      throw new Error(`data is not a string or buffer (${typeof data})`);
+    }
+    return data.includes(msg);
+  };
+}
+
+describe("CF3 and Extensions emulator", () => {
   let test: TriggerEndToEndTest;
 
   before(async function (this) {
     this.timeout(TEST_SETUP_TIMEOUT);
+    setUpExtensionsCache();
 
     expect(FIREBASE_PROJECT).to.exist.and.not.be.empty;
 
-    // TODO(joehan): Delete the --open-sesame call when extdev flag is removed.
-    const p = subprocess.spawnSync("firebase", ["--open-sesame", "extdev"], { cwd: __dirname });
-    console.log("open-sesame output:", p.stdout.toString());
-
-    test = new TriggerEndToEndTest(FIREBASE_PROJECT, EXTENSION_ROOT, readConfig());
-    await test.startExtEmulators([
-      "--test-params",
-      "test-params.env",
-      "--test-config",
-      TEST_CONFIG_FILE,
-    ]);
+    const config = readConfig();
+    const port = config.emulators!.storage.port;
+    process.env.STORAGE_EMULATOR_HOST = `http://localhost:${port}`;
+
+    test = new TriggerEndToEndTest(FIREBASE_PROJECT, __dirname, config);
+    await test.startEmulators();
+
+    admin.initializeApp({
+      projectId: FIREBASE_PROJECT,
+      credential: admin.credential.applicationDefault(),
+      storageBucket: `${FIREBASE_PROJECT}.appspot.com`,
+    });
   });
 
   after(async function (this) {
     this.timeout(EMULATORS_SHUTDOWN_DELAY_MS);
+    cleanUpExtensionsCache();
     await test.stopEmulators();
   });
 
-  it("should execute an HTTP function", async function (this) {
-    this.timeout(EMULATORS_SHUTDOWN_DELAY_MS);
+  it("should call a CF3 HTTPS function to write to the default Storage bucket", async function (this) {
+    this.timeout(EMULATOR_TEST_TIMEOUT);
+
+    const response = await test.writeToDefaultStorage();
+    expect(response.status).to.equal(200);
+
+    /*
+     * We delay again here because the functions triggered
+     * by the previous two writes run parallel to this and
+     * we need to give them and previous installed test
+     * fixture state handlers to complete before we check
+     * that state in the next test.
+     */
+    await new Promise((resolve) => setTimeout(resolve, EMULATORS_WRITE_DELAY_MS));
```

Comment:
So, this setTimeout chunk actually just blocks for EMULATORS_WRITE_DELAY_MS.

this.timeout() sets the total timeout for that test case - if the test exceeds that, it will fail.